lib/eventHubReceiver.ts

```diff
@@ -157,131 +172,18 @@ export class EventHubReceiver extends EventEmitter {
-        this._session = await rheaPromise.createSession(this._context.connection);
-        const handleReceiverError = (context: rheaPromise.Context) => {
-          receiverError = errors.translate(context.receiver.error);
-          debug(`An error occurred while creating the receiver "${this.name}" : `, receiverError);
-        };
-        this._session.on(Constants.receiverError, handleReceiverError);
-        debug("Trying to create a receiver...");
-        this._receiver = await rheaPromise.createReceiver(this._session, rcvrOptions);
-        debug("Promise to create the receiver resolved. Created receiver with name: ", this.name);
-        if (receiverError) {
-          // There are cases where the EH service sends an attach frame, which causes rhea to emit receiver_open event
-          // thus resolving the promise to create a receiver and moments later the service sends back a detach frame
-          // indicating that there was some error. Hence we check for receiverError, even after the promise has resolved.
-          debug("throwing the receiverError, ", receiverError);
-          throw receiverError;
-        }
-        this._session.removeListener(Constants.receiverError, handleReceiverError);
-        debug(`[${this._context.connectionId}] Receiver "${this.name}" created with receiver options: \n${JSON.stringify(rcvrOptions, undefined, 2)}`);
-      }
-      debug(`[${this._context.connectionId}] Negotatited claim for receiver "${this.name}" with with partition "${this.partitionId}"`);
-      this._ensureTokenRenewal();
-    } catch (err) {
-      if (err.value || (err.constructor && err.constructor.name === "c")) err = Errors.translate(err);
-      debug("Will reject the promise to create the receiver with error: %O", err);
-      throw (err);
-    }
-  }
-
-  /**
-   * Receive a batch of EventDatas from an EventHub partition for a given count and a given max wait time in seconds, whichever
-   * happens first.
+   * Receive a batch of EventData objects from an EventHub partition for a given count and a given max wait time in seconds, whichever
+   * happens first. This method can be used directly after creating the receiver object and **MUST NOT** be used along with the `start()` method.
    *
-   * @param {number} maxMessageCount                         The maximum message count. Must be a value greater than 0.
+   * @param {number} maxMessageCount                 The maximum message count. Must be a value greater than 0.
    * @param {number} [maxWaitTimeInSeconds]          The maximum wait time in seconds for which the Receiver should wait
    * to receiver the said amount of messages. If not provided, it defaults to 60 seconds.
    * @returns {Promise<EventData[]>} A promise that resolves with an array of EventData objects.
    */
-  async receive(maxMessageCount: number, maxWaitTimeInSeconds?: number): Promise<EventData[]> {
+  receive(maxMessageCount: number, maxWaitTimeInSeconds?: number): Promise<EventData[]> {
```

Comment:
>maxWaitTimeInSeconds [](start = 35, length = 20)

I don't think it's a problem considering how clearly the parameter is named, just want to point out that in many places node uses milliseconds so i'm not sure it's great to deviate from that. this being said, i've done it too so... your call